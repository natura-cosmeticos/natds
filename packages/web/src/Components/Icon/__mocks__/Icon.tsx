import * as React from 'react'
import { IMockedIconProps } from './Icon.props'

export const Icon : React.FunctionComponent<IMockedIconProps> = (props: IMockedIconProps) => <i {...props} />

export default Icon
