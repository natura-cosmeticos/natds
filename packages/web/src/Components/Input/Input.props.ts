import { InputProps } from '@material-ui/core/Input'
import { TextFieldSizes } from '../TextField/TextField.props'

export type Mask = Array<string | RegExp>;
export type MaskFn = () => Mask | void;
export type State = 'error' | 'success' | undefined;

type OmittedProps = 'classes' | 'inputComponent'

export interface IInputProps extends Omit<InputProps, OmittedProps> {

  /**
   * @optional
   * @type string
   */
  className?: InputProps['className']

  /**
   * If `true`, the `input` or `textarea` element will be disabled.
   *
   * @optional
   * @type bool
   */
  disabled?: InputProps['disabled']

  /**
   * If `true`, input will receive a padding-right equivalent to `large` spacing token
   *
   * @optional
   * @type bool
   */
  hasIcon?: boolean

  /**
   * The component used for the `input` element.
   * Either a string to use a HTML element or a component.
   *
   * @optional
   */
  inputComponent?: InputProps['inputComponent']

  /**
   * If `dense`, will adjust vertical spacing.
   *
   * This is normally obtained via context from `FormControl`.
   * The `normal` value is not supported.
   *
   * @type "dense" | "none"
   */
  margin?: InputProps['margin']

  /**
   * Mask format. Based on [React Input Mask](https://github.com/sanniassin/react-input-mask)
   *
   * @optional
   * @see https://github.com/sanniassin/react-input-mask
   */
  mask?: Mask | MaskFn

  /**
   * If `true`, a textarea element will be rendered.
   *
   * @default false
   * @optional
   * @type bool
   */
  multiline?: InputProps['multiline']

  /**
   * @optional
   * @type "error" | "success"
   */
  state?: State

  /**
   * The height of input field
   * @default mediumX
   */
  size?: TextFieldSizes
}
