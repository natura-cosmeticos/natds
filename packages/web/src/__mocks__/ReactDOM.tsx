// eslint-disable-next-line no-use-before-define
import * as React from 'react'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const MockReactDOM: any = jest.createMockFromModule('react-dom')

export const mockCreatePortal = jest.fn(
  (child, container) => (
    <div>
      <div id="content">{child}</div>
      <div id="target" data-target-tag-name={container.tagName} />
    </div>
  )
)

MockReactDOM.createPortal = mockCreatePortal

export { MockReactDOM }

export default MockReactDOM
