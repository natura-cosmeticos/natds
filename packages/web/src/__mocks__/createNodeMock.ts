const mockClientHeight = 100
const mockClientWidth = 300

// eslint-disable-next-line @typescript-eslint/no-explicit-any,max-lines-per-function
export const createNodeMock: (element: any) => (HTMLElement) = (element) => {
  if (typeof element.type !== 'undefined') {
    const mockElement: HTMLElement = document.createElement(element.type)

    mockElement.style.padding = '0'
    Reflect.defineProperty(mockElement, 'clientHeight', {
      get() {
        return mockClientHeight
      },
      enumerable: true,
      configurable: true
    })
    Reflect.defineProperty(mockElement, 'clientWidth', {
      get() {
        return mockClientWidth
      },
      enumerable: true,
      configurable: true
    })

    return mockElement
  }

  return null
}

export default createNodeMock
