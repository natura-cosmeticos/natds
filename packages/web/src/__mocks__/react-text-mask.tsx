import React from 'react';
import { MaskedInputProps } from 'react-text-mask';

export default function mock(props: MaskedInputProps) {
  const { render, ...otherProps } = props;

  function setRef() {
    return;
  }

  return props.render ?
    props.render(setRef, { ...otherProps }) :
    <input {...otherProps} />;
}
