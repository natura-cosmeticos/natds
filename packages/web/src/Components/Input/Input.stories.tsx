import { Meta, Story } from '@storybook/react'
import * as React from 'react'
import Input, { IInputProps } from './Input'

export default {
  component: Input,
  title: 'Utilities/Input'
} as Meta

const Template : Story<IInputProps> = (args: IInputProps) => <Input {...args} />

export const Playground : Story<IInputProps> = Template.bind({})
