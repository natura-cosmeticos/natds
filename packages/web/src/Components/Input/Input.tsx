import MaterialInput from '@material-ui/core/Input'
import * as React from 'react'
import { IInputProps } from './Input.props'

export { IInputProps } from './Input.props'

export const Input = React.forwardRef<unknown, IInputProps>(
  (props: IInputProps, ref: IInputProps['ref']) => {
    const {
      disabled,
      hasIcon,
      multiline,
      rows,
      state,
      ...otherProps
    } = props

    return (
      <MaterialInput
        {...otherProps}
        data-state={state}
        disabled={disabled}
        multiline={multiline}
        ref={ref}
        rows={rows}
        style={{ marginTop: 0 }}
      />
    )
  }
)
Input.displayName = 'Input'
export default Input
