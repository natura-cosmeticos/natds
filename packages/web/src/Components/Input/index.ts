import Input from './Input'

export { IInputProps } from './Input.props'
export { Input }
export default Input
