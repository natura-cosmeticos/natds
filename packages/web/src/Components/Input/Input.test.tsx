import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'
import { Input } from './Input'
import { createNodeMock } from '../../__mocks__/createNodeMock'

describe('Input component', () => {
  describe('when no props are provided', () => {
    it('should match snapshot', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <Input />, { createNodeMock }
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
  describe('when hasIcon property is true', () => {
    it('should match snapshot', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <Input hasIcon />, { createNodeMock }
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
  describe('when hasIcon property is false', () => {
    it('should match snapshot', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <Input hasIcon={false} />, { createNodeMock }
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
  describe('when state is error', () => {
    it('should match snapshot', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <Input state="error" />, { createNodeMock }
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
  describe('when state is success', () => {
    it('should match snapshot', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <Input state="success" />, { createNodeMock }
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
  describe('when multiline property is explicitly false', () => {
    it('should match snapshot', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <Input multiline={false} />, { createNodeMock }
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
  describe('when rows property is set', () => {
    it('should match snapshot', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <Input rows={1} />, { createNodeMock }
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
  describe('when rows property is set and multiline property is explicitly false', () => {
    it('should match snapshot', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <Input rows={1} />, { createNodeMock }
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
})
