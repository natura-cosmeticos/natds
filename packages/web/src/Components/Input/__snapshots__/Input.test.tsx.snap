// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component when hasIcon property is false should match snapshot 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline"
  onClick={[Function]}
>
  <input
    className="MuiInputBase-input MuiInput-input"
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`Input component when hasIcon property is true should match snapshot 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline"
  onClick={[Function]}
>
  <input
    className="MuiInputBase-input MuiInput-input"
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`Input component when multiline property is explicitly false should match snapshot 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline"
  onClick={[Function]}
>
  <input
    className="MuiInputBase-input MuiInput-input"
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`Input component when no props are provided should match snapshot 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline"
  onClick={[Function]}
>
  <input
    className="MuiInputBase-input MuiInput-input"
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`Input component when rows property is set and multiline property is explicitly false should match snapshot 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline"
  onClick={[Function]}
>
  <input
    className="MuiInputBase-input MuiInput-input"
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    rows={1}
    type="text"
  />
</div>
`;

exports[`Input component when rows property is set should match snapshot 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline"
  onClick={[Function]}
>
  <input
    className="MuiInputBase-input MuiInput-input"
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    rows={1}
    type="text"
  />
</div>
`;

exports[`Input component when state is error should match snapshot 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline"
  data-state="error"
  onClick={[Function]}
>
  <input
    className="MuiInputBase-input MuiInput-input"
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`Input component when state is success should match snapshot 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline"
  data-state="success"
  onClick={[Function]}
>
  <input
    className="MuiInputBase-input MuiInput-input"
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;
